packages/grafana-data/src/types/datasource.ts

```diff
@@ -473,6 +473,12 @@ export interface DataQueryResponse {
   traceIds?: string[];
 }
 
+export interface TestDataSourceResponse {
+  status: string;
+  // string in case of a success, Error object in case of error
+  message: string | Error;
```

Comment:
Nice that you are tidying that `any` up! 👏 

I think it would be semantically more correct to keep the message being just a "string", and have an optional property for the error itself. (Then we would get rid of the comment as well.)

```typescript
export interface TestDataSourceResponse {
    status: string;
    message: string;
    error?: Error;
}
```